File: cookbook/Csharp/Segments/createSegmentDynamic.cs
using Newtonsoft.Json.Linq;
﻿using System;
using System.Net;
using System.Xml;

namespace createSegmentDynamic
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            // Changez le Path pour correspondre à la destination de votre fichier de configuration
            var doc = new XmlDocument();
            doc.Load("config.xml");

            var xKey = doc.DocumentElement.SelectSingleNode("/config/xKey").InnerText;
            var baseUrl = doc.DocumentElement.SelectSingleNode("/config/url").InnerText;

            string segmentId = null; //Id du segment a modifier ('null' si le segment doit etre cree)

            var type = "dynamic"; //Code pour creer un segment Dynamique
            var name = "SegmentDynamic (Csharp)"; //Nom du segment
            var description = "SegmentDynamic (Csharp)"; //Description du segment
            var expiration = "2026-08-08T12:11:00Z"; //Date d'expiration du segment
            var isTest = true; //Le test est un segment de test : oui = 'true' / non = 'false'

            var url = baseUrl + "V1/segments/" + segmentId;

            //Creation du Json du message
            var jsonData = new JObject();
            jsonData.Add("type", type);
            jsonData.Add("name", name);
            jsonData.Add("description", description);
            jsonData.Add("expiration", expiration);
            jsonData.Add("isTest", isTest);

            //On affiche le message
            Console.Write(jsonData + "\n");


            //Lancement de la connexion pour remplir la requete
            var result = Utils.allConnection(url, xKey, jsonData, (segmentId == null ? "POST" : "PUT"));
            
            //Verification des reponses
            if ((int) result[0] == 200)
            {
                Console.Write("{0}", (string) result[3]);
            }
            else
            {
                //Affichage de l'erreur
                Console.Write("Error : {0} {1}", (int) result[0], ((HttpWebResponse) result[2]).StatusCode.ToString());
            }

            Console.ReadLine();
            ((HttpWebResponse) result[2]).Close();
        }
    }
}

File: cookbook/java/Segments/createSegmentDynamic.java
package Segments;

import java.io.FileInputStream;
import java.io.IOException;

import java.util.Map;
import java.util.Properties;

import org.json.JSONException;
import org.json.JSONObject;

import Utils.Request;

public class createSegmentDynamic 
{
    public static void main(String[] args) throws IOException, JSONException, InterruptedException
    {
        /*
        ** Renseignez ici les paramètres que vous souhaitez
        ** @var segmentId           => ID du segment à modifier (renseignez null pour le créer)
        ** @var type                => Type du segment ('static' ou 'dynamic')
        ** @var name                => Nom du segment
        ** @var description         => Description du segment
        ** @var expiration          => Date d'expiration du segment
        ** @var isTest              => Définit si le segment est un segment de test ou non (true: segment de test // false: n'est pas segment de test)
        ** @var parentId            => ID du segment parent (renseignez null s'il n'en a pas)
        */
        String segmentId    = null;
        String type         = "dynamic";
        String name         = "SegmentDynamic (java)";
        String description  = "SegmentDynamic (java)";
        String expiration   = "2026-01-08T12:11:00Z";
        Boolean isTest      = true;
        String parentId     = null;


        Properties properties = new Properties();
        try {
            properties.load(new FileInputStream("config.properties"));
        } catch (IOException e) {
            System.out.print(e.getMessage());
        }
        String xKey = properties.getProperty("xKey");
        String baseUrl = properties.getProperty("url");
        String url = baseUrl + "/V1/segments/";

        if (segmentId != null)
        {
            url += segmentId;
        }

        //Creation du Json du message
        JSONObject jsonData = new JSONObject();
        jsonData.put("type", type);
        jsonData.put("name", name);
        jsonData.put("description", description);
        jsonData.put("expiration", expiration);
        jsonData.put("isTest", isTest);
        jsonData.put("parentId", parentId);

        //On affiche le message
        System.out.print(jsonData + "\n\n");

        //Lancement de la connexion
        Map<String, String> resp;
        if (segmentId != null) {
            resp = Request.connection(url, xKey, jsonData, "PUT");
        }
        else {
            resp = Request.connection(url, xKey, jsonData, "POST");
        }

        //Verification des reponses
        if (resp.get("code").equals("200")) {
            //Le segment a bien ete modifie
            System.out.print("The segment dynamic  : " + name + " has been updated.");
        }
        else {
            //Affichage de l'erreur
            System.out.print("Error : " + resp.get("code") + " " + resp.get("reply"));
        }
    }
}
